core/src/auctions.ts

```diff
@@ -128,6 +130,34 @@ export const fetchAllAuctions = async function (network: string): Promise<Auctio
     return await Promise.all(auctionsWithMarketValue.map(a => enrichAuctionWithTransactionFees(a, fees)));
 };
 
+const enrichTakeEventWithDate = async function (network: string, takeEvent: TakeEvent): Promise<TakeEvent> {
+    const date = await fetchDateByBlockNumber(network, takeEvent.blockNumber);
+    return {
+        ...takeEvent,
+        date,
+    };
+};
+
+export const fetchTakeEvents = async function (network: string, auctionURL: string): Promise<Array<TakeEvent>> {
+    const { collateralType, auctionId } = parseAuctionURL(auctionURL);
+    const encodedAuctionId = convertNumberTo32Bytes(auctionId);
```

Comment:
I think there is a confusion between `auction.id`, `auction.auctionId` and `auctionId` that is revealed in this PR. I would suggest to rename type `auction.auctionId` to `auction.index` and then use those as equals: `auction.id` = `auctionId`, `auction.index` = `auctionIndex`. That would make many places less confusing:
```suggestion
    const { collateralType, auctionIndex } = parseAuctionURL(auctionId);
    const encodedAuctionIndex = convertNumberTo32Bytes(auctionIndex);
```
Would you think it makes sense to do it here or in another PR (merged before this one)?

Reply:
I would do it in a separate PR. It seems to be a bigger refracting job.

Reply:
Then please make another PR, so we can merge it before this PR is merged!

Reply:
I just committed it in here directly!